crates/uv-trampoline/src/bounce.rs

```diff
@@ -91,27 +109,82 @@ fn find_python_exe(is_gui: bool) -> CString {
                 break;
             }
         }
-        // Replace the filename (the last segment of the path) with "python.exe"
-        // Assumption: We are not in an encoding where a backslash byte can be part of a larger character.
-        let Some(last_backslash) = buffer.iter().rposition(|byte| *byte == b'\\') else {
-            eprintln!(
-                "Invalid current exe path (missing backslash): `{}`",
-                CString::from_vec_with_nul_unchecked(buffer)
-                    .to_string_lossy()
-                    .as_ref()
-            );
-            ExitProcess(1);
-        };
-        buffer.truncate(last_backslash + 1);
-        buffer.extend_from_slice(if is_gui {
-            b"pythonw.exe\0"
-        } else {
-            b"python.exe\0"
-        });
         CString::from_vec_with_nul_unchecked(buffer)
     }
 }
 
+/// The scripts are in the same directory as the Python interpreter, so we can find Python by getting the locations of
+/// the current .exe and replacing the filename with `python[w].exe`.
+fn find_python_exe(is_gui: bool, executable_name: &CStr) -> CString {
+    // Replace the filename (the last segment of the path) with "python.exe"
+    // Assumption: We are not in an encoding where a backslash byte can be part of a larger character.
+    let bytes = executable_name.to_bytes();
+    let Some(last_backslash) = bytes.iter().rposition(|byte| *byte == b'\\') else {
+        eprintln!(
+            "Invalid current exe path (missing backslash): `{}`",
+            &*executable_name.to_string_lossy()
+        );
+        unsafe {
+            ExitProcess(1);
+        }
+    };
+
+    let mut buffer = bytes[..last_backslash + 1].to_vec();
+    buffer.extend_from_slice(if is_gui {
+        b"pythonw.exe"
+    } else {
+        b"python.exe"
+    });
+    buffer.push(b'\0');
+
+    unsafe { CString::from_vec_with_nul_unchecked(buffer) }
+}
+
+fn push_arguments(output: &mut Vec<u8>) {
+    let arguments_as_str = unsafe { CStr::from_ptr(GetCommandLineA() as _) };
+
+    // Skip over the executable name and then push the rest of the arguments
+    let after_executable = skip_one_argument(arguments_as_str.to_bytes());
+
+    output.extend_from_slice(after_executable)
+}
+
+// TODO copy tests from MSDN for parsing
+fn skip_one_argument(arguments: &[u8]) -> &[u8] {
```

Comment:
I tried to write tests for this function but failed to get this crate to compile with `cargo test`.... I ended up coping the implementation into another crate and do some smoke testing (and I found and fixed a bug in the implementation)